classes/navigationMenu/NavigationMenuDAO.inc.php

```diff
@@ -275,6 +275,15 @@ function installSettings($contextId, $filename) {
 				$area = $navigationMenuNode->getAttribute('area');
 
 				$navigationMenu = null;
+
+				// Check if the given area has a NM attached.
+				// If it does the NM is not being processed and a warning is being thrown
+				$navigationMenusWithArea = $this->getByArea($contextId, $area)->toArray();
+				if (count($navigationMenusWithArea) != 0) {
+					error_log("WARNING: The NavigationMenu (ContextId: $contextId, Title: $title, Area: $area) will be skipped because the specified area has already a NavigationMenu attached.");
```

Comment:
Hmmm... The installer is functioning differently I think: See this line for example: https://github.com/pkp/pkp-lib/blob/master/classes/install/PKPInstall.inc.php#L82. That way the error message is set, and then here in the installer form considered: https://github.com/pkp/pkp-lib/blob/master/classes/install/form/InstallForm.inc.php#L203-L212 i.e. when using the CLI: https://github.com/pkp/pkp-lib/blob/master/classes/cliTool/InstallTool.inc.php#L79.

Reply:
Hmmm... but maybe it depends when and how this happens... maybe it would also be enough if there is the error in the server/PHP log file -- e.g. if the error is not a major one and the installation can live with it?

Reply:
When CLI is used, the logger is set, e.g. https://github.com/pkp/pkp-lib/blob/master/classes/cliTool/InstallTool.inc.php#L57. Thus the steps one goes through are displayed to the user, e.g. https://github.com/pkp/pkp-lib/blob/master/classes/install/Installer.inc.php#L412.
But I don't think this happens when Web UI is used for installation or upgrade.

Reply:
@defstat so my understanding is that when the CLI is being used, we should report this out. But otherwise no need to log it to the error log.

Reply:
Yes, correct, and this function error_log is then right, I would say -- it will print it for the user when using CLI (at least this is surely the case by an upgrade) and else it will print the message in the server error log file, I think.

Reply:
Ok great then no change needed! :)